shell/platform/common/cpp/text_input_model.h

```diff
@@ -47,6 +53,13 @@ class TextInputModel {
   // Returns true if any deletion actually occurred.
   bool Delete();
 
+  // Deletes text near the cursor.
+  //
+  // The offset is the number of characters relative to the cursor.
+  //
+  // Returns true if any deletion actually occurred.
+  bool DeleteSurrounding(int offset, int n_chars);
```

Comment:
This seems like an incredibly confusing API; I had to go read all the tests to figure out what exactly it did. I was going to suggest moving the logic up to the GTK code, but I guess you can't really do that because of surrogate pair handling. Some questions/comments:
- Saying that the offsets is "the number of characters" and then having `n_chars` as the name of an argument is extra confusing.
- What does "characters" mean? GIven the unicode handling, please use a non-ambiguous term like "code point" or "UTF-16 code unit" or whatever it is precisely.
- The implementation seems to be using code units; is that what the GTK API is expecting? I'm not familiar with the API, but you're passing the arguments straight through to here and I'm wondering if there's a measurement mismatch here.
- This says "cursor"; does that mean it should ignore selection? It would be good to make that explicit, and also have tests where there's a selection.

Reply:
It is a confusing API, I wouldn't be proposing it if the input methods didn't require it :/ Yeah, I looked at doing it in the GTK code but this seemed cleaner.

I've rewritten the documentation so this is hopefully clearer, please see if it makes more sense.

I believe the [GTK API](https://developer.gnome.org/gtk3/stable/GtkIMContext.html#GtkIMContext-delete-surrounding) is referring to code points, though I will dig deeper to confirm that.

Reply:
I think you have unpushed changes; I'm not seeing new documentation.

Reply:
Pushed now.